src/Microsoft.Azure.ServiceBus/Amqp/AmqpExceptionHelper.cs

```diff
@@ -71,17 +75,26 @@ public static AmqpResponseStatusCode GetResponseStatusCode(this AmqpMessage resp
             return responseStatusCode;
         }
 
-        public static Exception ToMessagingContract(Error error, bool connectionError = false)
+        public static Exception ToMessagingContractException(this AmqpMessage responseMessage, AmqpResponseStatusCode statusCode)
+        {
+            AmqpSymbol errorCondition = AmqpExceptionHelper.GetResponseErrorCondition(responseMessage, statusCode);
+            var statusDescription = responseMessage.ApplicationProperties.Map[ManagementConstants.Response.StatusDescription];
```

Comment:
Question: if we know that this can be `null`, why not to perform null coalesce right away? I.e.

```c#
var statusDescription = responseMessage.ApplicationProperties.Map[ManagementConstants.Response.StatusDescription] as string ?? errorCondition.Value;
```